### ISSUE TITLE
Expected assignment-from-no-return is not reported

### ISSUE THREAD

**Issue by shayts7** [PyCQA/pylint#2081]
### Steps to reproduce
1. Save this code:
```
class A(object):

  def inner(self):
    print('hi')

  def call_me(self):
    x = self.inner()
    print(x)


obj = A()
obj.call_me()
```
2. run `pylint -E <file>.py`

### Current behavior
No errors found

### Expected behavior
Expecting to get `assignment-from-no-return`

### pylint --version output
```
No config file found, using default configuration
pylint 1.7.5,
astroid 1.6.0
Python 2.7.12 (default, Nov 19 2016, 06:48:10)
[GCC 5.4.0 20160609]
```

**Comment by PCManticore** [comment]
Sounds like a bug!
